### PR TITLE
Add make target for custom-deploy

### LINKED ISSUE #152: [Instascale] Simplify image build, push, and deployment (all-in-one)
### Name of Feature or Improvement

Simplify the build, push, and deployment of a custom image of Instascale in a K8s cluster. 

### Description of Problem the Feature Should Solve

Having to manually run several commands to achieve the above for development purposes can be time-consuming.

### Describe the Solution You Would Like to See

There are several steps involved in achieving this. It would be best to have an all-in-one make target to simplify the process to use your own custom image of Instascale.

### PR DESCRIPTION
**Pull request by ChristianZaccaria**
# Issue link

Closes #152

The changes here are similar to what is done in this [PR for MCAD](https://github.com/project-codeflare/multi-cluster-app-dispatcher/pull/639)  

# What changes have been made

- The `custom-deploy` make target can be used for building and pushing a custom image of InstaScale and deploying it on your K8s cluster for development.
- Added documentation to Readme.

Note: using the name `custom-deploy` for the make target as there is already an existing easy way of deploying InstaScale (latest) to your K8s cluster. This make target focuses on the custom image deployments.

# Verification steps

Example: `make custom-deploy ENGINE=<podman or docker> IMG=quay.io/<username>/instascale:<image tag>`

## Checks
- [x] I've made sure the tests are passing. 
- Testing Strategy
   - [x] Manual tests

### CODE REVIEW

**Review by Fiona-Waters — COMMENTED** [review]
I have run the make command on an OSD cluster and can confirm that an `instascale-controller-manager` deployment and pod are created where I can view Instascale logs. I applied an appwrapper and can see that instascale picks it up as expected also.
Just one small comment regarding the command in make file. Otherwise it looks good.

File: Makefile [hunk 1]
```diff
@@ -115,6 +115,11 @@ endif
 install: manifests kustomize ## Install CRDs into the K8s cluster specified in ~/.kube/config.
 	$(KUSTOMIZE) build config/default | kubectl apply -f -
 
+# custom-deploy can be used for building and pushing a custom image of InstaScale and deploying it on your K8s cluster for development.
+# Example: "make custom-deploy ENGINE=<podman or docker> IMG=quay.io/<username>/instascale:<image tag>"
+.PHONY: custom-deploy
+easy-deploy: image-build image-push install deploy
```
[the remark this answers is not in the document]
Comment (ChristianZaccaria) [reply Makefile]: Excellent catch, thanks Fiona! That change is there now :+1:

[the remark this answers is not in the document]
Comment (Fiona-Waters) [reply Makefile]: /lgtm

**Comment by ChristianZaccaria** [comment]
> I have run the make command on an OSD cluster and can confirm that an `instascale-controller-manager` deployment and pod are created where I can view Instascale logs. I applied an appwrapper and can see that instascale picks it up as expected also. Just one small comment regarding the command in make file. Otherwise it looks good.

@Fiona-Waters Brilliant, thanks for testing it out!

**Comment by VanillaSpoon** [comment]
lgtm thanks @ChristianZaccaria

**Review by VanillaSpoon — APPROVED** [review]
lgtm thanks @ChristianZaccaria :D

**Review by anishasthana — APPROVED** [review]

**Pull request merged**